consensus/ibft/src/main/java/org/hyperledger/besu/consensus/ibft/statemachine/IbftRound.java

```diff
@@ -158,23 +168,41 @@ private boolean updateStateWithProposedBlock(final Proposal msg) {
     final boolean wasPrepared = roundState.isPrepared();
     final boolean wasCommitted = roundState.isCommitted();
     final boolean blockAccepted = roundState.setProposedBlock(msg);
+
     if (blockAccepted) {
+      final Block block = roundState.getProposedBlock().get();
+
+      final Signature commitSeal;
+      try {
+        commitSeal = createCommitSeal(block);
+      } catch (final SecurityModuleException e) {
+        LOG.warn("Failed to construct commit seal; {}", e.getMessage());
+        return blockAccepted;
+      }
+
       // There are times handling a proposed block is enough to enter prepared.
       if (wasPrepared != roundState.isPrepared()) {
         LOG.debug("Sending commit message. round={}", roundState.getRoundIdentifier());
-        final Block block = roundState.getProposedBlock().get();
-        transmitter.multicastCommit(getRoundIdentifier(), block.getHash(), createCommitSeal(block));
+        transmitter.multicastCommit(getRoundIdentifier(), block.getHash(), commitSeal);
+      }
+
+      // can automatically add _our_ commit message to the roundState
+      // cannot create a prepare message here, as it may be _our_ proposal, and thus we cannot also
+      // prepare
+      try {
+        final Commit localCommitMessage =
+            messageFactory.createCommit(
+                roundState.getRoundIdentifier(), msg.getBlock().getHash(), commitSeal);
+        roundState.addCommitMessage(localCommitMessage);
+      } catch (final SecurityModuleException e) {
+        LOG.warn("Failed to create signed Commit message; {}", e.getMessage());
+        return blockAccepted;
```

Comment:
Should we be returning true in this case, as this will result in a prepare message being created?

Reply:
From where this is called - the return value implies the block is good, and that this node should try and send a prepare message - which, regardless of this failure, is still true.